### ISSUE TITLE
Sending messages on 1.19.1 from offline player

### ISSUE THREAD

**Issue by Kapitowa** [PaperMC/Velocity#806]
Latest velocity, which support 1.19.1, without plugins.
Latest paper server without plugins.
`force-key-authentication = false` in velocity.toml
`enforce-secure-profile=false` in server.properties
Offline player can't send any message, it's just kicked.

Paper log is clear.
Error log from velocity on disconnect:
```
[17:37:30 INFO]: [connected player] Mr_White (/127.0.0.1:56789) has disconnected: An internal error occurred in your connection.
[17:37:30 ERROR]: [connected player] Mr_White (/127.0.0.1:56789): exception encountered in com.velocitypowered.proxy.connection.client.ClientPlaySessionHandler@7c8503e8
io.netty.handler.codec.CorruptedFrameException: Error decoding class com.velocitypowered.proxy.protocol.packet.chat.PlayerChat Direction SERVERBOUND Protocol 1.19.1 State PLAY ID 5
        at com.velocitypowered.proxy.protocol.netty.MinecraftDecoder.handleDecodeFailure(MinecraftDecoder.java:128) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at com.velocitypowered.proxy.protocol.netty.MinecraftDecoder.tryDecode(MinecraftDecoder.java:83) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at com.velocitypowered.proxy.protocol.netty.MinecraftDecoder.channelRead(MinecraftDecoder.java:58) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:327) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:299) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[velocity-3.1.2-SNAPSHOT-163.jar:3.1.2-SNAPSHOT (git-1a3fba42-b163)]
        at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: com.velocitypowered.proxy.util.except.QuietDecoderException: Incorrectly signed chat message
```

**Comment by astei** [comment]
Can't reproduce

**Comment by alisa12s** [comment]
https://www.youtube.com/watch?v=Qg2pJFm0kPo&feature=youtu.be

it's on a pirate account

**Comment by alisa12s** [comment]
> 


problem on video. 
try again with a pirate account

**Comment by astei** [comment]
I literally tried with a "pirate" account as stated above. It is working for me: https://youtu.be/wKhtN_Q8lkw

My server is plain Paper 1.19.1 and my client is completely unmodified vanilla 1.19.1.

**Comment by Kapitowa** [comment]
> I literally tried with a "pirate" account as stated above. It is working for me: https://youtu.be/wKhtN_Q8lkw
> 
> My server is plain Paper 1.19.1 and my client is completely unmodified vanilla 1.19.1.

I reproduced this on video, if you can check what's wrong I do: https://www.youtube.com/watch?v=5YpD1Uw32eE

**Comment by astei** [comment]
This is clearly a client issue caused by mods. What mods are you using?

**Comment by Kapitowa** [comment]
> This is clearly a client issue caused by mods. What mods are you using?

It's clear client 1.19.1 without mods (I am using for testing TL Legacy): https://youtu.be/s02P-oXH9dI

**Comment by astei** [comment]
I'm led to believe it's probably your "pirate" Minecraft launcher doing things it shouldn't be doing. I promise to look at it some more but it's low priority.

**Comment by Kapitowa** [comment]
I even downloaded PolyMC, which is recommended on the PaperMC discord. There is the same problem...
I would like to test your launcher, but this version only for linux. (but even so, the majority uses launchers not on Linux).
Some people report this issue also in discord, I think it's worth looking into what's wrong there.
If someone can figure out what the problem is here, because there are no mods, a clean client is launched from a pirate account.

**Comment by BBaoVanC** [comment]
Uh there is most definitely a Windows version of PolyMC. Could even try MultiMC since it should be compatible.

**Comment by Kapitowa** [comment]
MultiMC works with offline players. But before launching the launcher, I had to be sure to log in with a license, when I logged in, the necessary libraries were downloaded. Then I closed minecraft and opened it through a pirate account. 

But I still don't understand why everyone has to use MultiMC to play well on version 1.19.1 through a pirate account. If this problem did not occur before 1.19.1 with other launchers. Most players from my country who connect to the server use local launchers (TLauncher, TL Legacy), not MultiMC, and they get kicked regularly for sending chat messages.

**Comment by potatoru** [comment]
I have the same issue with a pirate account on TL Legacy.

**Comment by JNNGL** [comment]
Looks like the salt on 1.19.1 may be non-zero for unsigned messages

1.19.1:
<img src="https://cdn.discordapp.com/attachments/970670773939343360/1004009167876538458/unknown.png">

1.19:
<img src="https://cdn.discordapp.com/attachments/970670773939343360/1004009168220454992/unknown.png">

**Comment by astei** [comment]
fixed with #814
